src/main/java/eu/dissco/backend/service/SpecimenService.java

```diff
@@ -130,55 +135,55 @@ public JsonApiListResponseWrapper getDigitalMedia(String id, String path) {
   }
 
   public DigitalSpecimenJsonLD getSpecimenByIdJsonLD(String id) {
-    var digitalSpecimen = repository.getLatestSpecimenById(id);
+    var digitalSpecimenWrapper = repository.getLatestSpecimenById(id);
     var digitalMediaObjects = digitalMediaObjectService.getDigitalMediaIdsForSpecimen(
-        digitalSpecimen.id()).stream().map(value -> "hdl:" + value).toList();
-    var primarySpecimenData = generatePrimaryData(digitalSpecimen);
+            digitalSpecimenWrapper.digitalSpecimen().getOdsId()).stream().map(value -> "hdl:" + value)
+        .toList();
+    var primarySpecimenData = generatePrimaryData(digitalSpecimenWrapper);
     return new DigitalSpecimenJsonLD(
-        "hdl:" + digitalSpecimen.id(),
-        digitalSpecimen.type(),
+        "hdl:" + digitalSpecimenWrapper.digitalSpecimen().getOdsId(),
```

Comment:
Part of probably a bigger change. I think we are inconsistent with hdl and doi. I would propose to do DOI for digital specimen and media even though it might not all have been setup.

Reply:
https://naturalis.atlassian.net/browse/DD-760

Reply:
Also got this one to fix issues with json-ld. However, as long as prefixes are not fixed this is hard to do.
https://naturalis.atlassian.net/browse/DD-747